sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+* MUST use a .gitignore configuration
+* Source code repositories SHOULD follow the naming convention `[platform]-[library name]`, e.g. `android-analytics`, `java-datastore`, `ios-inapmessaging`, `swift-utils`
+    * Rationale: encoding version information (e.g. alpha/core/stable) in the repository breaks links and integration when a module changes
+    * Exceptions: existing modules that already carry additional information in the repository name & renaming would break too many existing integrations.
+* MUST NOT restrict read access to source code repositories unnecessarily
+* For internally hosted code → add the appropriate mailing list with read access
+* For open source projects → public repo on GitHub
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * SHOULD follow the standard project layout for [Android](https://developer.android.com/studio/projects) or for [Java](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_project_layout) projects respectively
+    * MUST prefix all exported resources (layout files, xml files, drawable files, color ids, dimension ids, string ids, etc.)  with <shortname> + "_" (e.g. "sug_" for a Suggestion Module) using [resourcePrefix](https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.LibraryExtension.html#com.android.build.gradle.LibraryExtension:resourcePrefix) in the gradle build
+* For iOS <img src="images/ios.png" alt="drawing" width="15"/>:
+    * SHOULD follow the standard project layout for [iOS](https://developer.apple.com/documentation/xcode/managing-files-and-folders-in-your-xcode-project) projects.
+    * RECOMMEND to not have source file header comment blocks (such as the Xcode default header comment) as these easily get out of date and the information such as author/version of commits is tracked in SCM
+
+
+### Build
+* Development version MUST build with single command after clean clone, i.e. no local setup necessary
+* MUST NOT depend on locally stored libraries
+* MUST NOT use wildcards (+) for exported dependencies, i.e. dependencies that SDK clients will inherit
+* MUST have a sample app - a minimal app that integrates the SDK and shows how to use the SDK features. It SHOULD be written in Kotlin (Android) or Swift (iOS).
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * MUST use [gradle](https://docs.gradle.org/current/userguide/userguide.html) with [gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) to build Android library or Java library
+    * MUST have consistent minSdk and supportLib versions with other SDKs
+        * SHOULD use [shared build configuration](https://github.com/rakutentech/android-buildconfig) for consistent version with other SDKs/libraries
+        * MAY use manual configuration with values (same as in [shared build configuration](https://github.com/rakutentech/android-buildconfig)), if so
+            * SHOULD use minSdk 23
```

Comment:
Maybe
> use a recent minSdk version unless there is a good reason not to

cos mentioning specific versions means this soon goes out of date

Reply:
changed to suggested statement